modules/solid_mechanics/src/actions/CommonSolidMechanicsAction.C

```diff
@@ -29,8 +31,98 @@ CommonSolidMechanicsAction::CommonSolidMechanicsAction(const InputParameters & p
 void
 CommonSolidMechanicsAction::act()
 {
+
   // check if sub-blocks block are found which will use the common parameters
-  auto action = _awh.getActions<QuasiStaticSolidMechanicsPhysicsBase>();
-  if (action.size() == 0)
+  auto actions = _awh.getActions<QuasiStaticSolidMechanicsPhysicsBase>();
+  if (actions.size() == 0)
     mooseWarning("Common parameters are supplied, but not used in ", parameters().blockLocation());
+
+  // Add disp-variables
+  if (_current_task == "add_variable")
+  {
+
+    // helper variables to collect what the user wants
+    bool do_add_variables = isParamSetByUser("add_variables") && getParam<bool>("add_variables");
+    std::vector<VariableName> displacement_variables;
+    if (isParamSetByUser("displacements"))
+    {
+      const auto v = getParam<std::vector<VariableName>>("displacements");
+      if (v.size() > 0)
+        displacement_variables.insert(displacement_variables.end(), v.begin(), v.end());
+    }
+    bool add_variables_block_restricted = true;
+    std::set<SubdomainName> add_variables_blocks;
+    bool scaling_set = isParamSetByUser("scaling");
+    Real scaling_value = (scaling_set) ? getParam<Real>("scaling") : 1.0;
+
+    // check all sub-actions
+    for (const auto & action : actions)
+    {
+      if (action->isParamSetByUser("add_variables") && action->getParam<bool>("add_variables"))
```

Comment:
I see that you are using `isParamSetByUser` a lot. I don't see them being necessary. Just add the default value (in this case a boolean false) to validParams and you only need to call `getParam`. The default value (false) will be retrieved if the parameter is not set by user. Similar suggestion applies elsewhere, unless I misunderstood your intent here (which is possible).

Reply:
I think your assessment is correct.